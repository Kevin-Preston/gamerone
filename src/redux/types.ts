import { AuthState } from './auth/types';
import { ProfileState } from './profile/types';
import { SettingsState } from './settings/types';
import { PostState } from './post/types';
import { PostFormState } from './post-form/types';
import { RequestStatusState } from './request-status/types';
import { RouterState } from 'connected-react-router';

export const INIT_STATE = 'root/INIT_STATE';
export const LOAD_STATE = 'root/LOAD_STATE';

export interface InitStateAction {
  type: typeof INIT_STATE;
}

export type RootStateActions = InitStateAction;

export interface RootState {
  Auth: AuthState;
  Profile: ProfileState;
  Settings: SettingsState;
  Post: PostState;
  PostForm: PostFormState;
  RequestStatus: RequestStatusState;
  router: RouterState;
}
