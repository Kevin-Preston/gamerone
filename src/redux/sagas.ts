import { all } from 'redux-saga/effects';
import authSagas from './auth/saga';
import profileSagas from './profile/saga';
import settingsSagas from './settings/saga';
import postSagas from './post/saga';
import postFormSagas from './post-form/saga';
import miscSagas from './misc/saga';

export default function* rootSaga() {
  yield all([
    authSagas(),
    profileSagas(),
    settingsSagas(),
    postSagas(),
    postFormSagas(),
    miscSagas(),
  ]);
}
