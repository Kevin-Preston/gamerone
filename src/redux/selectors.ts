/**
 * Here comes the combined selectors from different states
 */

import { createSelector } from 'reselect';

import { selectCurrentUser } from './auth/selectors';

import {
  selectCurrentProfileUser,
  selectCurrentProfileNetworks,
  selectCurrentProfileSponsors,
  selectCurrentProfileProducts,
} from './profile/selectors';

import {
  selectSettingsSponsors,
  selectSettingsProducts,
  selectSettingsProfileUser,
  selectProfileSocialNetworks,
} from './settings/selectors';

// determines whether the currently viewing profile is my profile or not
export const selectIsSelfProfile = createSelector(
  [selectCurrentProfileUser, selectCurrentUser],
  (profileUser, authUser) => authUser.id === profileUser?.id,
);

export const selectProfileUser = createSelector(
  [selectCurrentProfileUser, selectSettingsProfileUser],
  (profileUser, settingsUser) =>
    settingsUser.id === profileUser?.id ? settingsUser : profileUser,
);

// resolves social networks to be shown on Profile page
export const selectSocialNetworks = createSelector(
  [
    selectIsSelfProfile,
    selectProfileSocialNetworks,
    selectCurrentProfileNetworks,
  ],
  (isSelfProfile, settingsNetworks, profileNetworks) =>
    isSelfProfile ? settingsNetworks : profileNetworks,
);

// resolves sponsors to be shown on Profile page
export const selectSponsors = createSelector(
  [selectIsSelfProfile, selectSettingsSponsors, selectCurrentProfileSponsors],
  (isSelfProfile, settingsSponsors, profileSponsors) =>
    isSelfProfile ? settingsSponsors : profileSponsors,
);

// resolves products to be shown on Profile page
export const selectProducts = createSelector(
  [selectIsSelfProfile, selectSettingsProducts, selectCurrentProfileProducts],
  (isSelfProfile, settingsProducts, profileProducts) =>
    isSelfProfile ? settingsProducts : profileProducts,
);
