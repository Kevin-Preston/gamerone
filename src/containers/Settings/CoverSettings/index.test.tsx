import React from 'react';
import { Provider } from 'react-redux';
import { createProduct } from 'redux';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import authReducer from 'redux/auth/reducer';
import BannerSettings from './index';

const testUser = {
  id: 1449,
  username: 'vadim',
  email: 'vadimas.workteam@gmail.com',
  firstName: 'frontend',
  lastName: 'dev',
  bio: null,
  banner: 'profile/vadim.svg',
  avatar: null,
  playCount: 0,
  gameCount: 0,
  badgeCount: 0,
  followCount: 0,
  followerCount: 0,
  birthDate: null,
};

const renderWithAuthReducer = (
  component = <BannerSettings />,
  {
    initialState,
    store = createProduct(authReducer, { Auth: initialState }),
  } = {},
) => {
  let utils = {};
  act(() => {
    utils = render(<Provider store={store}>{component}</Provider>);
  });

  return utils;
};

describe('banner settings', () => {
  test('renders the form without crash', async () => {
    const { asFragment } = renderWithAuthReducer(<BannerSettings />, {
      initialState: { user: testUser },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  test('shows edit image modal on image load', async () => {
    renderWithAuthReducer(<BannerSettings />, {
      initialState: { user: testUser },
    });

    // TODO: Simulate file load event
  });

  test('disable the form button on load', () => {
    const { getByRole } = renderWithAuthReducer(<BannerSettings />, {
      initialState: { user: testUser },
    });

    expect(getByRole('button', { name: 'Upload' }).disabled).toEqual(true);
  });

  test('renders the user banner on load', () => {
    const { getByLabelText } = renderWithAuthReducer(<BannerSettings />, {
      initialState: { user: testUser },
    });

    expect(
      getByLabelText(/banner/).style.backgroundImage.includes(testUser.banner),
    ).toBe(true);
  });
});
