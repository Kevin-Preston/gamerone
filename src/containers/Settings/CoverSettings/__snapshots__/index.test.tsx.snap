// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner settings renders the form without crash 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <form>
      <div
        class="card__content"
      >
        <h4>
          Cover
        </h4>
        <div
          class="dropzone"
          tabindex="0"
        >
          <input
            accept="image/*"
            aria-label="file-upload"
            autocomplete="off"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div
            aria-label="banner"
            style="height: 110px; background-image: url(https://cdn.gamerone.gg/1100x220/profile/vadim.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
          />
        </div>
      </div>
      <div
        class="card__actions"
      >
        <button
          class="button button--primary"
          disabled=""
          type="submit"
        >
          Upload
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
